packages/fast-tooling-react/src/css-editor/border/border.tsx

```diff
@@ -0,0 +1,127 @@
+        prevState: CSSBorderState
+    ): void {
+        if (this.state !== prevState) {
+            this.props.onChange({
+                ...this.state
+            });
+        }
+    }
+
+    public render(): React.ReactNode {
+        return (
+            <div className={get(this.props, "managedClasses.cssBorder")}>
+                <label className={get(this.props, "managedClasses.cssBorder_label")}>
+                    Border
+                </label>
+                <div className={get(this.props, "managedClasses.cssBorder_control")}>
+                    <div
+                        className={get(
+                            this.props,
+                            "managedClasses.cssBorder_colorInputRegion"
+                        )}
+                        style={{ background: this.state.borderColor }}
+                    >
+                        <input
+                            type={"color"}
+                            style={{ opacity: 0, width: "100%" }}
+                            value={this.state.borderColor}
+                            onChange={this.handleBorderColorOnChange}
+                        />
+                    </div>
+                    <span
+                        className={get(
+                            this.props,
+                            "managedClasses.cssBorder_selectControl"
+                        )}
+                    >
+                        <select
+                            className={get(this.props, "managedClasses.cssBorder_select")}
+                            onChange={this.handleBorderStyleOnChange}
+                            value={this.state.borderStyle}
+                        >
+                            {Object.keys(BorderStyleValue).map(this.renderBorderOption)}
+                        </select>
+                    </span>
+                    <input
+                        className={get(this.props, "managedClasses.cssBorder_input")}
+                        type={"text"}
+                        placeholder={"1px"}
+                        value={this.state.borderWidth}
+                        onChange={this.handleBorderWidthOnChange}
+                    />
+                </div>
+            </div>
+        );
+    }
+
+    private renderBorderOption(style: string): JSX.Element {
+        return (
+            <option key={style} value={style}>
+                {style
+                    .charAt(0)
+                    .toUpperCase()
+                    .concat(style.slice(1))}
+            </option>
+        );
+    }
+
+    private handleBorderStyleOnChange = (
+        e: React.ChangeEvent<HTMLSelectElement>
+    ): void => {
+        this.setState({
+            borderStyle: e.target.value as BorderStyleValue,
+        });
+    };
+
+    private handleBorderColorOnChange = (
+        e: React.ChangeEvent<HTMLInputElement>
+    ): void => {
+        this.setState({
+            borderColor: e.target.value,
```

Comment:
per my comment above, you will want to change this to `e.target.value || this.defaultBorderColor` same with all the other handlers setting state for these values.